hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoBatchSqlConnectorTest.java

```diff
@@ -0,0 +1,358 @@
+
+        createMapping(true);
+        collection.insertOne(new Document("firstName", "Han").append("lastName", "Solo").append("jedi", "false"));
+        collection.insertOne(new Document("firstName", "Anakin").append("lastName", "Skywalker").append("jedi", "true"));
+
+        assertRowsAnyOrder("select firstName, lastName, jedi from " + methodName()
+                        + " where lastName = ?",
+                singletonList("Skywalker"),
+                asList(
+                        new Row("Luke", "Skywalker", true),
+                        new Row("Anakin", "Skywalker", true)
+                )
+        );
+    }
+
+    @Test
+    public void readsWithJoins() {
+        final String connectionString = mongoContainer.getConnectionString();
+
+        MongoCollection<Document> peopleName = database.getCollection("peopleName");
+        peopleName.insertOne(new Document("personId", 1).append("name", "Luke Skywalker"));
+        peopleName.insertOne(new Document("personId", 2).append("name", "Han Solo"));
+
+        MongoCollection<Document> peopleProfession = database.getCollection("peopleProfession");
+        peopleProfession.insertOne(new Document("personId", 1).append("profession", "Jedi"));
+        peopleProfession.insertOne(new Document("personId", 2).append("profession", "Smuggler"));
+
+        execute("CREATE MAPPING peopleName external name \"peopleName\" (personId INT, name VARCHAR) "
+                + "TYPE MongoDB "
+                + "OPTIONS ("
+                + "    'connectionString' = '" + connectionString + "', "
+                + "    'database' = '" +  databaseName + "'"
+                + ")");
+        execute("CREATE MAPPING peopleProfession external name \"peopleProfession\" (personId INT, profession VARCHAR) "
+                + "TYPE MongoDB "
+                + "OPTIONS ("
+                + "    'connectionString' = '" + connectionString + "', "
+                + "    'database' = '" + databaseName + "'"
+                + ")");
+
+        assertRowsAnyOrder("select pn.personId, pn.name, pr.profession " +
+                        "from peopleName pn join peopleProfession pr on pn.personId = pr.personId",
+                emptyList(),
+                asList(
+                        new Row(1, "Luke Skywalker", "Jedi"),
+                        new Row(2, "Han Solo", "Smuggler")
+                )
+        );
+    }
+
+    @Test
+    public void insertsIntoMongo_parametrized_withId() {
+        insertsIntoMongo(true, "insert into " + methodName() + "(jedi, firstName, lastName) values (?, 'Han', ?)",
+                false, "Solo");
+    }
+    @Test
+    public void insertsIntoMongo_hardcoded_withoutId() {
+        insertsIntoMongo(false, "insert into " + methodName() + "(jedi, firstName, lastName) " +
+                "values (false, 'Han', 'Solo')");
+    }
+
+    public void insertsIntoMongo(boolean includeId, String sql, Object... args) {
```

Comment:
`insertsIntoMongo` does not only insert but also assert. But the SQL you pass can do anything and the assert won't match.

split it into 2 and the test will do 2 clear steps:
insert...
assert...

Reply:
The idea is that it's quasi-parametrized test - I have insertsIntoMongo as a base with some parameters and I call with from test methods with various valid set of parameters.

Maybe simple rename to `testInsertIntoMongo` will be sufficient?

Reply:
Ok, this makes more sense.